Cargo.toml

```diff
@@ -88,9 +88,18 @@ opt-level = 3
 embassy-executor = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
 embassy-hal-internal = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
 embassy-nrf = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+embassy-net = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+# No direct changes there, but required by embassy-net
+embassy-sync = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+embassy-time = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+embassy-time-driver = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+embassy-time-queue-driver = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
+embassy-net-driver = { git = "https://github.com/kaspar030/embassy", branch = "for-riot-rs-100424" }
 
 usbd-hid-macros = { git = "https://github.com/kaspar030/usbd-hid", branch = "for-riot-rs" }
 
+smoltcp.git = "https://github.com/smoltcp-rs/smoltcp.git" # version 0.12-to-be
```

Comment:
can we pin this to a specific ref?

Reply:
Yes we can:
```suggestion
# version 0.12-to-be
smoltcp = { git = "https://github.com/smoltcp-rs/smoltcp.git", rev = "125773e282bc2e0c914a49e9c75573926e26e558" }
```

I didn't go that route initially, because we do have a lock file, and any update to that part of the lockfile would then just fix the fallout, but if you prefer, this is the line to apply.

Reply:
yes please. might be me but I find the lockfile hard to manage at this scale (read: I let cargo manage it completely but then I don't know what `git = "https://github.com/smoltcp-rs/smoltcp.git", rev = 125773e282bc2e0c914a49e9c75573926e26e558" }` should point to)